service/interpreter/workflowImpl.go

```diff
@@ -782,10 +794,74 @@ func processStateExecution(
 		stateExeId, state, stateExecutionLocal, commandReq,
 		completedTimerCmds, completedSignalCmds, completedInterStateChannelCmds,
 	)
+
+	// Wait for decider trigger (ANY/ALL command completed) OR continue-as-new threshold
 	_ = provider.Await(ctx, func() bool {
 		return IsDeciderTriggerConditionMet(commandReq, completedTimerCmds, completedSignalCmds, completedInterStateChannelCmds) || continueAsNewCounter.IsThresholdMet()
 	})
+
 	commandReqDoneOrCanceled = true
```

Comment:
this is a tricky & key variable. you may leave some comments here ;)

Reply:
Added comments